tests/unit/forklift/test_legacy.py

```diff
@@ -1887,6 +1887,211 @@ def test_upload_fails_with_too_large_file(self, pyramid_config, db_request):
             "See /the/help/url/ for more information."
         )
 
+    def test_upload_fails_with_too_large_project_size_default_limit(
+        self, pyramid_config, db_request
+    ):
+        pyramid_config.testing_securitypolicy(userid=1)
+
+        user = UserFactory.create()
+        db_request.user = user
+        EmailFactory.create(user=user)
+        project = ProjectFactory.create(
+            name="foobar",
+            upload_limit=(60 * 1024 * 1024),  # 60 MB
+            # Default total_size_limit(legacy.MAX_PROJECT_SIZE)
+            total_size=legacy.MAX_PROJECT_SIZE - 1,  # 10 GB - 1
+        )
+        release = ReleaseFactory.create(project=project, version="1.0")
+        RoleFactory.create(user=user, project=project)
+
+        filename = "{}-{}.tar.gz".format(project.name, release.version)
+
+        db_request.POST = MultiDict(
+            {
+                "metadata_version": "1.2",
+                "name": project.name,
+                "version": release.version,
+                "filetype": "sdist",
+                "md5_digest": "nope!",
+                "content": pretend.stub(
+                    filename=filename,
+                    file=io.BytesIO(b"a" * 2),
+                    type="application/tar",
+                ),
+            }
+        )
+        db_request.help_url = pretend.call_recorder(lambda **kw: "/the/help/url/")
+
+        with pytest.raises(HTTPBadRequest) as excinfo:
+            legacy.file_upload(db_request)
+
+        resp = excinfo.value
+
+        assert db_request.help_url.calls == [pretend.call(_anchor="project-size-limit")]
+        assert resp.status_code == 400
+        assert resp.status == (
+            "400 Project size too large."
+            + " Limit for project 'foobar' total size is 10 GB. "
+            "See /the/help/url/"
+        )
+
+    def test_upload_fails_with_too_large_project_size_custom_limit(
+        self, pyramid_config, db_request
+    ):
+        pyramid_config.testing_securitypolicy(userid=1)
+
+        user = UserFactory.create()
+        db_request.user = user
+        EmailFactory.create(user=user)
+        one_megabyte = 1 * 1024 * 1024
+        project = ProjectFactory.create(
+            name="foobar",
+            upload_limit=(60 * 1024 * 1024),  # 60 MB
+            total_size=legacy.MAX_PROJECT_SIZE,  # 10 GB
```

Comment:
I think it'd be better to omit the "# 10 GB" comments, since these comments would be out-of-date/wrong if we change `legacy.MAX_PROJECT_SIZE`.